StoreKitUnitTests/StoreProductTests.swift

```diff
@@ -27,48 +36,46 @@ class StoreProductTests: StoreKitConfigTestCase {
             "com.revenuecat.annual_39.99.2_week_intro",
             "lifetime"
         ])
-        let sk1Fetcher = ProductsFetcherSK1(productsRequestFactory: ProductsRequestFactory(),
-                                            requestTimeout: Self.requestTimeout)
-        let sk1StoreProduct = try await sk1Fetcher.products(withIdentifiers: productIdentifiers)
-        let sk1StoreProductsByID = sk1StoreProduct.reduce(into: [:]) { partialResult, wrapper in
-            partialResult[wrapper.productIdentifier] = wrapper
-        }
+        let sk1StoreProducts = try await self.sk1Fetcher.products(withIdentifiers: productIdentifiers)
+        let sk1StoreProductsByID = sk1StoreProducts.uniqueIndex { $0.productIdentifier }
 
-        let sk2Fetcher = ProductsFetcherSK2()
-        let sk2StoreProduct = try await sk2Fetcher.products(identifiers: productIdentifiers)
-        let sk2StoreProductsByID = sk2StoreProduct.reduce(into: [:]) { partialResult, wrapper in
-            partialResult[wrapper.productIdentifier] = wrapper
-        }
+        let sk2StoreProducts = try await ProductsFetcherSK2().products(identifiers: productIdentifiers)
+        let sk2StoreProductsByID = sk2StoreProducts.uniqueIndex { $0.productIdentifier }
 
-        expect(sk1StoreProduct.count) == productIdentifiers.count
-        expect(sk1StoreProduct.count) == sk2StoreProduct.count
+        expect(sk1StoreProducts.count) == productIdentifiers.count
+        expect(sk1StoreProducts.count) == sk2StoreProducts.count
 
         for sk1ProductID in sk1StoreProductsByID.keys {
             let sk1Product = try XCTUnwrap(sk1StoreProductsByID[sk1ProductID])
             let equivalentSK2Product = try XCTUnwrap(sk2StoreProductsByID[sk1ProductID])
 
-            expect(sk1Product.productIdentifier) == equivalentSK2Product.productIdentifier
-            expect(sk1Product.localizedDescription) == equivalentSK2Product.localizedDescription
-            expect(sk1Product.price) == equivalentSK2Product.price
-            expect(sk1Product.localizedPriceString) == equivalentSK2Product.localizedPriceString
-            expect(sk1Product.productIdentifier) == equivalentSK2Product.productIdentifier
-            expect(sk1Product.isFamilyShareable) == equivalentSK2Product.isFamilyShareable
-            expect(sk1Product.localizedTitle) == equivalentSK2Product.localizedTitle
-            if sk1Product.subscriptionGroupIdentifier != nil {
-                expect(sk1Product.subscriptionGroupIdentifier) == equivalentSK2Product.subscriptionGroupIdentifier
-            } else {
-                expect(equivalentSK2Product.subscriptionGroupIdentifier).to(beNil())
-            }
+            expectEqualProducts(sk1Product, equivalentSK2Product)
         }
     }
 
+    func testSK1AndStoreProductDetailsAreEquivalent() async throws {
```

Comment:
I had a bug in the `StoreProduct` implementation that this caught (`localizedDescription` was returning `localizedTitle`).